sfusurge/surge_website
# Contentful CMS integration

## Describe your changes
This PR integrates the Contentful CMS into the code. The official contentful Javascript client was used. Code for the join page and [jobId] dynamic route page was re-organized.

## Testing steps 
- checkout to `CMS`
- run `pnpm install` 
- update your local environment file with the the `CONTENTFUL_SPACE_ID` and `CONTENTFUL_ACCESS_TOKEN` environment variables, which can be found in the notion documentation in the technology department
- launch the local server, and check to ensure that all of the job listings appear in the `/join` route, and that clicking on an individual post leads to a dynamic route with the full job description
- Login to contentful (credentials in notion), create a new Job Listing entry, and then check to see that the entry appears on your local server. 

Once testing is complete and the correct behaviour is verified, please remove any dummy data created in contentful.

## Issue ticket number and link
#66 

## Checklist before requesting a review
